syft/federated/plan.py

```diff
@@ -185,18 +194,43 @@ def find_location(self, args):
 
     def copy(self):
         """Creates a copy of a plan."""
-        plan = Plan(sy.ID_PROVIDER.pop(), self.owner, self.name)
-        plan.blueprint = self.blueprint
-        plan.readable_plan = self.readable_plan
+        plan = Plan(
+            sy.ID_PROVIDER.pop(),
+            self.owner,
+            self.name,
+            arg_ids=self.arg_ids,
+            result_ids=self.result_ids,
+            readable_plan=self.readable_plan,
+            blueprint=self.blueprint,
+        )
+        plan.replace_ids(
+            from_ids=plan.arg_ids, to_ids=plan.arg_ids, from_worker=self.id, to_worker=plan.id
+        )
+        plan.replace_ids(
+            from_ids=plan.result_ids, to_ids=plan.result_ids, from_worker=self.id, to_worker=plan.id
+        )
         return plan
 
-    def replace_ids(self, from_ids: List[Union[str, int]], to_ids: List[Union[str, int]]):
+    def replace_ids(
+        self,
+        from_ids: List[Union[str, int]],
+        to_ids: List[Union[str, int]],
+        from_worker: Union[str, int] = None,
+        to_worker: Union[str, int] = None,
+    ):
         """Replaces pairs of tensor ids in the plan stored.
 
         Args:
-            from_ids: the left part of the pair: ids to change.
-            to_ids: the right part of the pair: ids to replace with.
+            from_ids: The left part of the pair: ids to change.
```

Comment:
The right and left part of the pair are unclear. Can you clarify this morning/just remove that part

Reply:
I propose to remove the reference to pairs. There are two lists and no pairs. Maybe check that the lists have the same length.